docs/errors/error_1000.md

```diff
@@ -0,0 +1,55 @@
+# Error code: 1000
+
+**Category:** High Bandwidth Memory (HBM) Out of Memory (OOM)
+
+**Type:** Compile time
+
+**XLA Backends:** TPU
+
+## Error log examples
+
+```
+RESOURCE_EXHAUSTED: XLA:TPU compile permanent error. Ran out of memory in memory space hbm. Used 49.34G of 32.00G hbm. Exceeded hbm capacity by 17.34G.
+```
+```
+RESOURCE_EXHAUSTED: Compilation failure: Aborting compilation early because it's unlikely to have enough memory. Requires 3.84T, has 90.96G available. If more detailed logging is desired, set --xla_tpu_impure_oom_fast_exit_threshold=-1
+```
+
+## Why do these happen?
+
+The XLA compiler throws these errors when the total High Bandwidth Memory (HBM) required at any specific point in the compiled program exceeds the physical capacity of the TPU device.
+
+To effectively debug these errors, it is crucial to understand that XLA allocates HBM for more than just your model weights. The available HBM is shared by a number of different functions, including:
+* **Parameters & Optimizer State:** The static weights, embeddings, and optimizer slot variables (e.g., momentum, variance)
+* **Intermediate Values:** Dynamic memory required for activation tensors, gradients, and temporary buffers during the forward/backward pass
+* **Program Code:** The actual executable binary of the compiled XLA program
+* **System Overhead:** A small portion of memory is reserved for XLA Runtime internals
+
+This error is thrown if the XLA compiler is unable to fit all of the above allocations on the device HBM.
+
+## How can a user fix their program when they do happen?
```

Comment:
This and the following section heading is currently the same as other error docs for consistency.

However, I think these can be simplified to:
* "How to fix this error?" OR "Potential fixes"
* "How to debug this error?" OR "Debugging techniques"

Reason: [Google documentation style guide](https://developers.google.com/style/person) prefers second person, and in our particular case, _you_ is implied.

If this sounds good, I'm happy to open a new PR to address this for all error code docs together, so that this PR isn't blocked.